Controllers/BrandsController.cs

```diff
@@ -0,0 +1,123 @@
+﻿using EcommerceStore.Controllers.InputModels;
+using EcommerceStore.Controllers.ViewModels;
+using EcommerceStore.Data.Context;
+using EcommerceStore.Data.Entities;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EcommerceStore.Controllers
+{
+    [ApiController]
+    [Route("api/brands")]
+    public class BrandsController : ControllerBase
+    {
+        private readonly EcommerceContext _context;
+
+        public BrandsController(EcommerceContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet("{brandId}")]
+        public async Task<ActionResult<BrandViewModel>> GetByIdAsync([FromRoute] int brandId)
+        {
+            var brandViewModel = await _context.Brands
+                .Where(b => b.Id == brandId)
+                .Select(b => new BrandViewModel
+                {
+                    Name = b.Name,
+                    FoundationYear = b.FoundationYear,
+                    ProductsCount = b.Products.Count()
+                })
+                .FirstAsync();
+
+            if (brandViewModel == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(brandViewModel);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<BrandViewModel>>> GetAllAsync()
+        {
+            var viewBrands = await _context.Brands
```

Comment:
It is better to use the following naming convension : {subject}InputModel or {subject}ViewModel
also you can use shortcuts Vm and Im
e.g. brandsViewModel or brandsVm